review08/README.md

```diff
@@ -0,0 +1,241 @@
+- If the database is connected.
+- If other dependencies are connected.
+- Check if external services are available.
+- Resource usage (CPU, memory, network,...).
+
+#### Types
+
+1. Liveness -- The liveness probe checks if the service is healthy. If the probe fails, the service is not running, cannot be recovered, and should be restarted.
+2. Readiness -- The readiness probe checks if the application is ready to serve traffic; requests are not prohibited. Dependencies, databases, and external services are connected. If the readiness probe fails, the service is not yet ready to serve traffic, but it's not considered a failure, and the service is not restarted. This probe prevents routing to the instance which is warming up or loading data.
+3. Startup -- The startup probe determines if the application considers itself successfully initialized. That doesn't necessarily mean it's ready to serve traffic, but rather that all core components and configurations are loaded and initialized.
+
+### SmallRye Health
+
+SmallRye Health is an implementation of MicroProfile Health. It simplifies the process of creating health checks and provides a way to check the state of the application. Due to the built-in health checks, it also exposes some of this information out-of-the-box.
+
+1. `HealthCheck` interface -- Health checks are implemented as classes that implement the `HealthCheck` interface.
+2. `@Liveness` annotations -- Checks if the service is running.
+3. `@Readiness` annotations -- Checks if the service is connected to the database/Kafka,... without any additional code and is ready to serve requests.
+4. `@Startup` annotations -- Option for slow starting containers to delay an invocation of the liveness check.
+
+## Monitoring and Metrics
+
+Observability is a key feature of the service. It's important to monitor what is happening inside the application to be able to react to the problems and to improve the performance.
+
+### What do we want to measure?
+
+- **Uptime** -- How long the service is running.
+- **Rate of requests** -- How many requests are coming to the service.
+- **Latency** -- How long it takes to process the request.
+- **Error rate** -- How many requests are failing.
+- **Memory usage** -- How much memory is used.
+- **CPU usage** -- How much CPU is used.
+- **Network usage** -- How much data is sent and received.
+- **Custom metrics** -- Business-specific metrics.
+
+### Custom metrics
+
+Business specific metrics are important to understand the state of the application. For example, how many passengers are being onboarded, how much baggage is lost, etc.
+
+## Tools for metrics and monitoring
+
+![Monitoring architecture](./img/metrics_architecture.jpg)
+
+
+### Micrometer metrics
+
+Micrometer metrics provide a way to collect metrics and send them to the monitoring system. With the `micrometer-registry-prometheus` extension, we can send metrics to Prometheus, such as memory usage, CPU usage, latency, custom metrics, etc.
```

Comment:
yeah, and what is Prometheus? How does it relate to micrometer? Shooting this kind of terms without context is confusing to students.

Reply:
You need to do at least a short intro when you introduce a new term. I see that the following section is prometheus. Here it should be just Prometheus (metrics registry).

But still, what is Micrometer?